File: AtCoder/ACP/BootCamp4b/Medium100_2/cpp/A_Diverse_Word.cpp
#include <bits/stdc++.h>
using namespace std;

string S;
int used[26];

void solve() {
  string res = "";
  int N = S.size();
  if (N < 26) {
    for (auto s : S) {
      used[s - 'a'] = true;
    }
    for (int i = 0; i < 26; ++i) {
      if (!used[i]) {
        res = S + string(1, i + 'a');
        break;
      }
    }
  } else {
    int i = N - 2;
    vector<char> V{S[N - 1]};
    while (i >= 0) {
      for (auto v : V) {
        if (S[i] < v) {
          res = S.substr(0, i) + v;
          printf("%s\n", res.c_str());
          return;
        }
      }
      V.push_back(S[i--]);
      sort(V.begin(), V.end());
    }
    res = "-1";
  }
  printf("%s\n", res.c_str());
}

int main() {
  cin >> S;
  solve();
}

File: AtCoder/ACP/BootCamp4b/Medium100_2/python/A_Diverse_Word.py
S = input()
N = len(S)
if N < 26:
    used = [False] * 26
    for s in S:
        used[ord(s) - ord('a')] = True
    for i in range(26):
        if not used[i]:
            res = S + chr(i + ord('a'))
            break
else:
    V = [S[N - 1]]
    for i in range(N - 2, -1, -1):
        for v in V:
            if S[i] < v:
                res = S[:i] + v
                print(res)
                exit()
        V.append(S[i])
        V.sort()
        i -= 1
    res = -1
print(res)
